Code/RDGeneral/RDValue-taggedunion.h

```diff
@@ -161,42 +161,52 @@ namespace RDTypeTag {
   }
 }
 
+const unsigned int KEYMAX = 0x00FFFFFF;
+
 struct RDValue {
   RDTypeTag::detail::Value value;
-  short type;
-  short reserved_tag; // 16 bit alignment
+  unsigned int type:8;
```

Comment:
I might be (actually less) clever by doing 

```
unsigned int key;
unsigned char type;
```

And only loose 8/16 bits as opposed to 32...